functions/src/index.ts

```diff
@@ -44,14 +44,23 @@ const handleCreateDocument = async (
   timestamp: Number
 ) => {
   try {
-    const data = await extract(snapshot, timestamp);
-
-    logs.debug({
-      ...data
-    });
-
-    logs.createIndex(snapshot.id, data);
-    await index.partialUpdateObject(data, { createIfNotExists: true });
+    const forceDataSync = config.forceDataSync;
+    if (forceDataSync === 'yes') {
+      const updatedSnapshot = await snapshot.ref.get();
+      const data = await extract(updatedSnapshot, 0);
+      logs.updateIndex(updatedSnapshot.id, data);
+      logs.info('force sync data: execute saveObject');
+      await index.saveObject(data);
```

Comment:
if we think saveObject is better in some cases, why isn't it used in all cases and simply a change? What advantage can partialUpdateObject bring still?

Reply:
This approach makes an additional call to the Firestore db.  I was trying to avoid making this extra call so that why I used partialUpdate with versioning.  However, when a field is removed, partialUpdate did not handle it.  Also, I didn't want to rip partial update out if it was working for customers.